source/MaterialXCore/Node.h

```diff
@@ -162,6 +162,8 @@ class Node : public InterfaceElement
     static const string CATEGORY;
 };
 
+using FlattGraphPredicate = std::function<bool(NodePtr node)>;
```

Comment:
It might be better to name this alias NodePredicate and defined it in Node.h, since it's not specific to graph flattening functionality.